pkg/k8shandler/service.go

```diff
@@ -97,34 +104,38 @@ func createOrUpdateService(serviceName, namespace, clusterName, targetPortName s
 	addOwnerRefToObject(service, owner)
 
 	err := client.Create(context.TODO(), service)
-	if err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("Failure constructing %v service: %v", service.Name, err)
-		}
+	if err == nil {
+		return nil
+	}
+	if !apierrors.IsAlreadyExists(err) {
+		return kverrors.Wrap(err, "failed to construct service",
+			"service_name", service.Name)
+	}
 
-		current := service.DeepCopy()
-		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			if err = client.Get(context.TODO(), types.NamespacedName{Name: current.Name, Namespace: current.Namespace}, current); err != nil {
-				if errors.IsNotFound(err) {
-					// the object doesn't exist -- it was likely culled
-					// recreate it on the next time through if necessary
-					return nil
-				}
-				return fmt.Errorf("Failed to get %v service: %v", service.Name, err)
+	current := service.DeepCopy()
```

Comment:
i realize we're just outdenting, but instead of a DeepCopy can we use an empty object here?